ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -239,15 +239,15 @@ let resync_host ~__context ~host =
               (Ref.string_of self) ;
             Xapi_clustering.Daemon.enable ~__context ;
             maybe_switch_cluster_stack_version ~__context ~self ~cluster_stack ;
+            let verify = Stunnel_client.get_verify_by_default () in
+            set_tls_config ~__context ~self ~verify ;
             (* Note that join_internal and enable both use the clustering lock *)
             Client.Client.Cluster_host.enable ~rpc ~session_id ~self
           ) ;
```

Comment:
There is an implicit `else` branch here for the "joined and not enabled" case, which now excludes `set_tls_config` as well. This is correct?

Reply:
I think this is a good point. If the cluster daemon is not enabled, then in theory xapi-clusterd service itself should not be running, hence any requests to xapi-clusterd should fail. That means the observer initialisation should be moved into the enabled path as well. Need to verify that....

Reply:
I believe there is no need to do anything if xapi-clusterd is not enabled, I have resturctured the code for this